### DISCUSSION TITLE
Rewards tracking for multi-agent

### DISCUSSION THREAD

**Discussion by berttggg** [Toni-SM/skrl#128] [Q&A]
I am reading the code of rewards tracking for multi-agents. 

And it seems that we are only interested in the first key's value, meaning the first agent only. Why? Or Anything I have missed?
Thank you. 

https://github.com/Toni-SM/skrl/blob/7b090a1674b3ff1240a49385bd7fd757571f7145/skrl/multi_agents/torch/base.py#L305

**Comment by Toni-SM** [comment]
Hi @berttggg 

Sorry for late response.

Yes, the current implementation uses only the reward from the first agent.

It was implemented using the Bi-DexHands environments for testing. Those environments define only one reward for all the task like Isaac Gym:
https://github.com/PKU-MARL/DexterousHands/blob/99c1e2a399fb084df5c02dbb5f6182d394fcd2e8/bidexhands/tasks/hand_base/base_task.py#L60-L61

In the next releases this implementation will be extended to a more generic form.

**Comment by berttggg** [comment]
I see, may I ask so far, the MAPPO or IPPO has any successful examples? How is the learning performance for the Bi-DexHands task?
